AppSyncRealTimeClient/Support/CountdownTimer.swift

```diff
@@ -0,0 +1,60 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+/// A resettable timer that executes `onCountdownComplete` closure after
+/// `interval`.
+///
+/// The timer will execute the closure on a background queue. If the closure
+/// includes work that must be performed on a specific queue, make sure to dispatch
+/// it inside the closure.
+class CountdownTimer {
+    /// The interval after which the timer will fire
+    let interval: TimeInterval
+
+    private var workItem: DispatchWorkItem?
```

Comment:
Any thoughts on using DispatchSourceTimer? the `autoFlush` timer in analytics plugin uses DispatchSourceTimer, however its use case is for a recurring timer. Haven't tried it myself to see if this completely fits the use case, but saw there is a repeat interval of .never to make it only fire once: https://developer.apple.com/documentation/dispatch/dispatchsourcetimer/2920392-schedule

Reply:
Mostly because `DispatchWorkItem` is what I thought of first. :) (I originally started down the road of using a straight-up `Timer`, but run loop management is a bit fraught.) I'll play with DispatchSourceTimer, but unless I see a clear win, I'll probably stick with the `DispatchWorkItem`, since at the very least the API for creating one is simpler.

Reply:
I played around with it a bit but don't see a compelling case to use it instead of DispatchWorkItem, since we're not repeating this, but rather pre-empting it when we receive new data